app/scripts/controllers/deployment.js

```diff
@@ -260,6 +269,28 @@ angular.module('openshiftConsole')
           Logger.log("builds (subscribe)", $scope.builds);
         }));
 
+        function updateFilterWarning() {
+          if (!LabelFilter.getLabelSelector().isEmpty() && $.isArray($scope.replicaSetsForDeployment) && !$scope.replicaSetsForDeployment.length && !$.isEmptyObject($scope.unfilteredReplicaSetsForDeployment)) {
```

Comment:
Lodash makes it a little easier because you can pass an array or object to `_.isEmpty`

```js
if (!LabelFilter.getLabelSelector().isEmpty() &&
     _.isEmpty(scope.replicaSetsForDeployment) &&
    !_.isEmpty($scope.unfilteredReplicaSetsForDeployment)) {
  // ...
}
```